packages/flutter_tools/lib/src/base/net.dart

```diff
@@ -29,7 +29,11 @@ Future<List<int>> fetchUrl(Uri url) async {
   }
 }
 
-Future<List<int>> _attempt(Uri url) async {
+/// Check if the given URL points to a valid endpoint.
+Future<bool> doesRemoteFileExist(Uri url) async =>
+  (await _attempt(url, true)) != null;
+
+Future<List<int>> _attempt(Uri url, bool onlyHeaders) async {
```

Comment:
Sorry, one more thing: Use a named parameter for the bool so that the call site self-documents what the "true" means:

```suggestion
Future<List<int>> _attempt(Uri url, {bool onlyHeaders = false}) async {
```

Reply:
Done.